.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,42 @@
+---
+name: Build & upload PyPI package
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  build-package:
+    name: Build & verify package
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: hynek/build-and-inspect-python-package@v1
+
+  release-pypi:
+    name: Publish released package to pypi.org
+    environment: release-pypi # TODO: how to name the env
```

Comment:
Maybe also `Deployment` like we have now?

Reply:
`Deployment` seems like a sensible choice to me. 👍